msal/src/test/java/com/microsoft/identity/client/robolectric/shadows/ShadowStrategyResultServerError.java

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+package com.microsoft.identity.client.robolectric.shadows;
+
+import com.microsoft.identity.common.internal.net.HttpResponse;
+import com.microsoft.identity.common.internal.net.ObjectMapper;
+import com.microsoft.identity.common.internal.providers.oauth2.TokenErrorResponse;
+import com.microsoft.identity.internal.testutils.strategies.MockTestStrategy;
+
+import org.robolectric.annotation.Implements;
+
+@Implements(MockTestStrategy.class)
+public class ShadowStrategyResultServerError {
```

Comment:
Question : Do we need to create each individual Shadow for every server error if we want to test ?

 for example errors from server like  `BAD_TOKEN`, `Protection_Policy_Required` etc ?

Reply:
With the current state of the tests, yes. There is WIP on moving away from that. Of course the developer will still need create a mock success or error response but we're moving away from doing that in the shadow and will provide some other mechanism to inject different mocked responses into the test instead of using the shadow